packages/components/src/calendar-react-day-picker/index.tsx

```diff
@@ -0,0 +1,210 @@
+				chevron: `${ BASE_CLASSNAME }__chevron`,
+				nav: `${ BASE_CLASSNAME }__nav`,
+				month_caption: `${ BASE_CLASSNAME }__month-caption`,
+				months: `${ BASE_CLASSNAME }__months`,
+				month_grid: `${ BASE_CLASSNAME }__month-grid`,
+				weekday: `${ BASE_CLASSNAME }__weekday`,
+				today: `${ BASE_CLASSNAME }__day--today`,
+				selected: `${ BASE_CLASSNAME }__day--selected`,
+				disabled: `${ BASE_CLASSNAME }__day--disabled`,
+				hidden: `${ BASE_CLASSNAME }__day--hidden`,
+				range_start: `${ BASE_CLASSNAME }__range-start`,
+				range_end: `${ BASE_CLASSNAME }__range-end`,
+				range_middle: `${ BASE_CLASSNAME }__range-middle`,
+				weeks_before_enter: `${ BASE_CLASSNAME }__weeks-before-enter`,
+				weeks_before_exit: `${ BASE_CLASSNAME }__weeks-before-exit`,
+				weeks_after_enter: `${ BASE_CLASSNAME }__weeks-after-enter`,
+				weeks_after_exit: `${ BASE_CLASSNAME }__weeks-after-exit`,
+				caption_after_enter: `${ BASE_CLASSNAME }__caption-after-enter`,
+				caption_after_exit: `${ BASE_CLASSNAME }__caption-after-exit`,
+				caption_before_enter: `${ BASE_CLASSNAME }__caption-before-enter`,
+				caption_before_exit: `${ BASE_CLASSNAME }__caption-before-exit`,
+			},
+			// Localization
+			locale,
+			formatters: {
+				formatWeekdayName: ( date: Date ) => {
+					// ie. M, T, W, T, F, S, S
+					return new Intl.DateTimeFormat( localeCode, {
+						weekday: 'narrow',
+					} ).format( date );
+				},
+				formatCaption: ( date: Date ) => {
+					// ie. April 2025
+					return new Intl.DateTimeFormat( localeCode, {
+						year: 'numeric',
+						month: 'long',
+					} ).format( date );
+				},
+			},
+			// a11y
+			role: 'application',
+		} as const;
+	}, [ numberOfMonths, locale ] );
+
+	return commonProps;
+};
+
+export const DateCalendar = ( {
+	[ 'aria-label' ]: ariaLabel = 'Date calendar',
+	locale,
+	numberOfMonths = 1,
+	defaultSelected,
+	selected: selectedProp,
+	onSelect,
+	...props
+}: DateCalendarProps ) => {
+	const commonProps = useCommonProps( { numberOfMonths, locale } );
+
+	const [ selected, setSelected ] = useControlledValue( {
+		defaultValue: defaultSelected,
+		value: selectedProp,
+		onChange: onSelect,
+	} );
+
+	return (
+		<DayPicker
+			aria-label={ ariaLabel }
+			{ ...props }
+			mode="single"
+			selected={ selected }
+			onSelect={ setSelected }
+			{ ...commonProps }
+		/>
+	);
+};
+
+export const DateRangeCalendar = ( {
+	[ 'aria-label' ]: ariaLabel = 'Date range calendar',
+	locale,
+	numberOfMonths = 1,
+	defaultSelected,
+	selected: selectedProp,
+	onSelect,
+	...props
+}: DateRangeCalendarProps ) => {
+	const commonProps = useCommonProps( { numberOfMonths, locale } );
+
+	const [ selected, setSelected ] = useControlledValue( {
+		defaultValue: defaultSelected,
+		value: selectedProp,
+		onChange: onSelect,
+	} );
+
+	const [ hoveredDate, setHoveredDate ] = useState< Date | undefined >( undefined );
+
+	// Compute the preview range for hover effect
+	const previewRange = useMemo( () => {
```

Comment:
Not sure if it is handled in this function or not, but for the date range when selecting a single day there is a discrepancy between the visual state and the `onSelect` callback ( see video ).
- If I click once, the `onSelect` returns the single date for both `from` and `to`
- If I click a second time on the same date, it shows selected but `onSelect` returns `undefined`
- If I click a third time it does deselect and `onSelect` returns `undefined`.
It would be great if for the second click we still return the range with `from` and `to` being the same date.

https://github.com/user-attachments/assets/480224d1-366e-4389-a1cf-e780e68ff744

Reply:
I'll need to investigate this one, thank you for raising

Reply:
I'm about to push some changes (ie passing `null` instead of `undefined` when using the component in controlled mode) , could you take a look again? I can't repro anymore on my machine, but good to double-check.